inat_finder.py

```
@@ -0,0 +1,589 @@
+        batch_end_time = time.time()
+
+        current_batch_time = batch_end_time - batch_start_time
+        batch_times.append(current_batch_time)
+        if len(batch_times) > max_batch_times_to_average:
+            batch_times.pop(0) # Keep only the last N times
+
+        if show_progress and pbar:
+            average_batch_time = sum(batch_times) / len(batch_times)
+            if average_batch_time > 0: # Avoid division by zero if a batch was instant
+                remaining_items = total_variations - pbar.n - len(batch) # pbar.n is updated after pbar.update()
+                if remaining_items < 0: remaining_items = 0 # Ensure non-negative
```

Ruff (0.11.9)

Multiple statements on one line (colon)

(E701)